### PR TITLE
Adding LifeCycleChange issue

### PR DESCRIPTION
**Pull request by HiranyaKavishani**
### Purpose
This PR will fix the issue in updating LifeCycleState

### Issues

Fixes https://github.com/wso2/product-microgateway/issues/1955

### Automation tests
 - Unit tests added: Yes/No
 - Integration tests added: Yes/No

### Tested environments

Not Tested

---
#### Maintainers: Check before merge
- [x] Assigned 'Type' label
- [x] Assigned the project
- [x] Validated respective github issues
- [ ] Assigned milestone to the github issue(s)

### CODE REVIEW

File: api/wso2/discovery/subscription/api.proto [hunk 1]
```diff
@@ -13,7 +13,7 @@ option java_multiple_files = true;
 
 // APIs data model
 message APIs {
-    string apiId = 1;
+  string apiId = 1;
```
Comment (Rajith90) [inline api/wso2/discovery/subscription/api.proto:16]: Formatting issue

File: adapter/internal/messaging/notification_listener.go [hunk 2]
```diff
@@ -219,14 +219,10 @@ func handleLifeCycleEvents(data []byte) {
...
 			for i := range apiListOfEnv {
-				if apiEvent.UUID == apiListOfEnv[i].UUID && (apiListOfEnv[i].APIStatus == blockedStatus ||
-					apiEvent.APIStatus == blockedStatus) {
-					//If previous or current state is 'Blocked' only we update the xds. All other states are neglected at the gateway
-					logger.LoggerMsg.Infof("Lifecycle state changed from %s to %s", apiListOfEnv[i].APIStatus, apiEvent.APIStatus)
-					apiListOfEnv[i].APIStatus = apiEvent.APIStatus
-					xds.UpdateEnforcerAPIList(configuredEnv, xds.MarshalAPIList(eh.APIListMap[configuredEnv]))
-					break
-				}
+				logger.LoggerMsg.Infof("Lifecycle state changed from %s to %s", apiListOfEnv[i].APIStatus, apiEvent.APIStatus)
```
Comment (Rajith90) [inline adapter/internal/messaging/notification_listener.go:222]: `if apiEvent.UUID == apiListOfEnv[i].UUID` check should be there

File: enforcer/src/main/java/org/wso2/choreo/connect/enforcer/api/APIFactory.java [hunk 3]
```diff
@@ -86,6 +87,19 @@ public void addApis(List<Api> apis) {
...
 
+    public void updateApis(List<APIs> apisList) {
+        for (APIs api : apisList) {
+            for (Map.Entry<String, API> entry : apis.entrySet()) {
+                String apiKey = entry.getKey();
+                API updatedAPI = entry.getValue();
+                if (updatedAPI.getAPIConfig().getUuid().equals(api.getUuid())) {
+                    updatedAPI.getAPIConfig().setApiLifeCycleState(api.getLcState());
+                }
+                apis.put(apiKey, updatedAPI);
```
Comment (Rajith90) [inline enforcer/src/main/java/org/wso2/choreo/connect/enforcer/api/APIFactory.java:98]: Where are we removing the auth filter when lc state is prototype

File: api/wso2/discovery/subscription/api.proto (hunk 1 above)
Comment (HiranyaKavishani) [reply api/wso2/discovery/subscription/api.proto:16]: Fixed the formatting issue :-)

**Comment by Rajith90** [comment]
Closing this PR since this does not handle all the cases

**Pull request closed without merging**